### PR TITLE
LPS-174794 ES8 Compatibility

### PR DESCRIPTION
**Pull request by BryanEngler**
Stories:
[LPS-176393](https://issues.liferay.com/browse/LPS-176393) Remove Document Types
[LPS-176395](https://issues.liferay.com/browse/LPS-176395) Add compatibility headers
[LPS-176398](https://issues.liferay.com/browse/LPS-176398) Gracefully handle Additional Types and Overriding

Subtasks:
[LPS-183776](https://issues.liferay.com/browse/LPS-183776) Remove Document Types
[LPS-184194](https://issues.liferay.com/browse/LPS-184194) Add compatibility headers
[LPS-184327](https://issues.liferay.com/browse/LPS-184327) Gracefully handle Additional Types and Overriding

Authors: @joshuacords, @peerkar, @lipusz, @BryanEngler

### CODE REVIEW

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:relevant

**Comment by BryanEngler** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
ci:test:search-solr

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:relevant

**Comment by BryanEngler** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
ci:test:search-solr

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:relevant

**Comment by BryanEngler** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
ci:test:search-solr

**Comment by BryanEngler** [comment]
ci:test:sf

**Comment by BryanEngler** [comment]
ci:test:search

**Comment by BryanEngler** [comment]
ci:test:search-solr

**Comment by BryanEngler** [comment]
ci:test:commerce-es7-remote

**Comment by BryanEngler** [comment]
ci:test:workflow-es7-remote

**Comment by BryanEngler** [comment]
ci:test:commerce

**Comment by BryanEngler** [comment]
ci:test:search-remote

**Comment by BryanEngler** [comment]
ci:test:search-remote-minimum

**Comment by BryanEngler** [comment]
ci:test:search-lxc

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/elasticsearch/Elasticsearch7.testcase [hunk 1]
```diff
@@ -64,7 +64,7 @@ definition {
 				-d '{
 					"query": {
 						"match": {
-							"_type": "LiferayDocumentType"
+							"_type": "_doc"
```
Comment (BryanEngler) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/elasticsearch/Elasticsearch7.testcase:67]: @xbrianlee let me know if you think there are any issues related to this test case. if there are issues i think we can probably just change it from a match query to a match_all query and not have to deal with the type

**Comment by joshchong** [comment]
@BryanEngler I aborted all the running tests in this PR since there are test setup changes that need to be made which we'll be testing in our own PRs against your branch.

**Comment by xbrianlee** [comment]
ci:forward

**Comment by xbrianlee** [comment]
ci:forward:force

**Pull request closed without merging**